### PR TITLE
CHANGELOG.md: Update for 23.11rc2

### PR DESCRIPTION
**Pull request by nepet**
Update changelog and py versions for second release candidate that includes mainly some tests for cln-rest

### CODE REVIEW

**Comment by endothermicdev** [comment]
the changed section should include the cln prefix added to clnrest config options. The full changes are:
```
config option `rest-certs` changed to `clnrest-certs`
config option `rest-protocol` changed to `clnrest-protocol`
config option `rest-host` changed to `clnrest-host`
config option `rest-port` changed to `clnrest-port`
config option `rest-cors-origins` changed to `clnrest-cors-origins`
config option `rest-csp` changed to `clnrest-csp`
```
Could probably be summarized in a single line as
clnrest config options `rest-certs`, `rest-protocol`, `rest-host`, `rest-port`, `rest-cors-origins`, `rest-csp` are all now prefixed with `cln` in order to avoid clashing with c-lightning-REST. (i.e., `rest-port` to `clnrest-port`) ([#6857])

**Comment by endothermicdev** [comment]
Is incrementing the pyln versioning necessary for rc2?  I might prefer leaving them as 23.11 for simplicity.

**Comment by nepet** [comment]
Thanks for your feedback @endothermicdev. I'll work it in and let you know once I am done!

File: CHANGELOG.md [hunk 1]
```diff
@@ -111,6 +115,9 @@ Note: You should always set `allow-deprecated-apis=false` to test for changes.
 [#6135]: https://github.com/ElementsProject/lightning/pull/6135
 [#6311]: https://github.com/ElementsProject/lightning/pull/6311
 [#6734]: https://github.com/ElementsProject/lightning/pull/6734
+[#6850]: https://github.com/ElementsProject/lightning/pull/6850
+[#6867]: https://github.com/ElementsProject/lightning/pull/6867
+[#6857]: https://github.com/ElementsProject/lightning/pull/6875
```
Comment (endothermicdev) [inline CHANGELOG.md:120]:
```suggestion
[#6857]: https://github.com/ElementsProject/lightning/pull/6857
```

**Comment by endothermicdev** [comment]
ACK 749b694afbcb8a7c5606c53d0979aa53cd23ee9d

File: CHANGELOG.md (hunk 1 above)
Comment (nepet) [reply CHANGELOG.md]: Wow! nice that you caught my human shortcomings! It would have been embarrassing if that got merged :sweat_smile:

**Comment by nepet** [comment]
Added https://github.com/ElementsProject/lightning/pull/6876

**Comment by nepet** [comment]
Added #6840 to the release candidate.

**Pull request merged**